src/main/java/core/basesyntax/reader/CsvFileReader.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.reader;
+
+import com.opencsv.CSVParser;
+import com.opencsv.CSVParserBuilder;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.exceptions.CsvValidationException;
+import core.basesyntax.exceptions.FileWasNotReadException;
+import core.basesyntax.handler.TransactionHandler;
+import core.basesyntax.handler.impl.BalanceTransactionHandler;
+import core.basesyntax.handler.impl.PurchaseTransactionHandler;
+import core.basesyntax.handler.impl.ReturnTransactionHandler;
+import core.basesyntax.handler.impl.SupplyTransactionHandler;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class CsvFileReader {
+    private final String filePath;
+    private final Map<String, TransactionHandler> transactionHandlers = new HashMap<>();
+
+    public CsvFileReader(String filePath) {
+        this.filePath = filePath;
+        transactionHandlers.put("b", new BalanceTransactionHandler());
+        transactionHandlers.put("s", new SupplyTransactionHandler());
+        transactionHandlers.put("p", new PurchaseTransactionHandler());
+        transactionHandlers.put("r", new ReturnTransactionHandler());
+    }
+
+    public Map<String, Integer> readTransactions() {
```

Comment:
Don't keep all logic in a single package. Consider moving CsvFileReader to a package that better represents its purpose, such as `service` or `service.impl`.